src/objects-table/ObjectsTable.js

```diff
@@ -248,15 +268,63 @@ class ObjectsTable extends React.Component {
         return this.actions.isContextActionAllowed(this.props.d2, ...args);
     };
 
+    getSelectionMessages = () => {
+        const { allObjects, dataRows, selection, pager } = this.state;
+        const messages = [];
+
+        const selectedAllInCurrentPage =
+            !_.isEmpty(dataRows) && dataRows.every(row => selection.has(row.id));
+        const selectedAllInAllPages = selection.size === pager.total;
+        const selectInAllPagesImplemented = allObjects.size === pager.total;
+
+        if (selectedAllInCurrentPage && !selectedAllInAllPages && selectInAllPagesImplemented) {
+            messages.push({
+                message: i18n.t("All {{count}} items on this page are selected.", {
+                    count: dataRows.length,
+                }),
+                link: i18n.t("Select all {{count}} items in all pages", { count: pager.total }),
+                action: this.selectAllPages,
+            });
+        }
+
+        const selectionInOtherPages = _.difference(
+            Array.from(selection),
+            dataRows.map(dr => dr.id)
+        );
+
+        if (selectedAllInAllPages) {
+            messages.push({
+                message: i18n.t("There are {{count}} items selected in all pages.", {
+                    count: pager.total,
+                }),
+                link: i18n.t("Clear selection"),
+                action: this.clearSelection,
+            });
+        } else if (selectionInOtherPages.length > 0) {
+            messages.push({
+                message: i18n.t("There are {{count}} items selected on other pages.", {
+                    count: selectionInOtherPages.length,
+                }),
+                link: i18n.t("Clear selection"),
+                action: this.clearSelection,
+            });
+        }
```

Comment:
It's ok as it is, no worries :) But just as a note, to make it functional, you may write:

```
const messages = _.compact([
  condition && message, // or condition ? message : null
  condition2 && message2,
  ...
])```